syntax = "proto2";
package config;

enum DefendingSide {
    AUTOREF_SIDE = 0;
    NEGATIVE = 1;
    POSITIVE = 2;
}

enum FriendlyColor {
    AUTOREF_COLOR = 0;
    BLUE = 1;
    YELLOW = 2;
}

message Backend {
    required string ssl_vision_ip = 1;
    required uint32 ssl_vision_port = 2;
    required string gamecontroller_ip = 3;
    required uint32 gamecontroller_port = 4;
}

message Perception {
    required DefendingSide defending_side = 1;
    required FriendlyColor friendly_color = 2;
}

message Gameplay {
    required bool div_a = 1;
}

message GuiBridge {
    required string unix_socket_prefix = 1;
    required string ssl_vision_topic = 2;
    required string ssl_gc_topic = 3;
    required string log_topic = 4;
    required string world_topic = 5;
}

message Rules {
    required float robot_max_radius = 1;
    required float ball_max_radius = 2;
    required uint32 max_robot_id = 3;
    required float max_kick_speed = 4;
    required float div_a_total_field_x_length = 5;
    required float div_a_total_field_y_length = 6;
    required float div_b_total_field_x_length = 7;
    required float div_b_total_field_y_length = 8;
}

message Config {
    required Backend backend = 1;
    required Perception perception = 2;
    required Gameplay gameplay = 3;
    required GuiBridge gui_bridge = 4;
    required Rules rules = 5;
}
